### PR TITLE
Homework hibernate

### PR DESCRIPTION
**Pull request by gujaKhela**

### CODE REVIEW

File: src/main/java/com/example/Ticket.java [hunk 1]
```diff
@@ -1,25 +1,44 @@
...
-    private final long creationTime = Instant.now().toEpochMilli();  // Creation time set at object instantiation
-    private TicketType ticketType; // Add this line
-    private int userId;
-    private int id;  // Auto-generated by DB
-    private long creationDate;
+
 
     public Ticket() {
+        this.creationDate = new Timestamp(System.currentTimeMillis());
         NullableWarningChecker.check(this);
```
Comment (Alexmansar) [inline src/main/java/com/example/Ticket.java]: after NullableWarningChecker.check(this);

File: src/main/java/com/example/TicketDAO.java [hunk 2]
```diff
@@ -1,54 +1,69 @@
...
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.EntityManagerFactory;
+import jakarta.persistence.Persistence;
+
 
 public class TicketDAO {
-    private static final String SAVE_TICKET_SQL = "INSERT INTO Ticket (user_id, ticket_type) VALUES (?, ?::ticket_type)";
-    private static final String FETCH_TICKET_SQL = "SELECT * FROM Ticket WHERE id = ?";
-    private static final String UPDATE_TICKET_SQL = "UPDATE Ticket SET ticket_type = ?::ticket_type WHERE id = ?";
+    private EntityManagerFactory emf = Persistence.createEntityManagerFactory("my_persistence");
```
Comment (Alexmansar) [inline src/main/java/com/example/TicketDAO.java]: could be final?

File: src/main/java/com/example/TicketDAO.java [hunk 3]
```diff
@@ -1,54 +1,69 @@
...
+        } catch (Exception e) {
+            if (em.getTransaction().isActive()) {
+                em.getTransaction().rollback();
+            }
+            System.err.println("Error saving ticket: " + e.getMessage());
+            e.printStackTrace();
+        } finally {
+            em.close();
         }
     }
```
Comment (Alexmansar) [inline src/main/java/com/example/TicketDAO.java]: try to use  try with resources

**Review by Alexmansar — CHANGES_REQUESTED** [review]
@gujaKhela Hi) In general good job, but I have some general comments. So please fix in all places)

File: src/main/java/com/example/Ticket.java (hunk 1 above)
Comment (gujaKhela) [reply src/main/java/com/example/Ticket.java]: i did not get it fully?

File: pom.xml [hunk 4]
```diff
@@ -8,19 +8,30 @@
...
-                <groupId>org.postgresql</groupId>
-                <artifactId>postgresql</artifactId>
-                <version>42.7.4</version>
-            </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.13.2</version>
+            <scope>test</scope>
+        </dependency>
```
Comment (Alexmansar) [inline pom.xml]: why we need it now?

File: src/main/java/com/example/Ticket.java (hunk 1 above)
Comment (Alexmansar) [reply src/main/java/com/example/Ticket.java]: I mean - first validation, if all OK, than this.creationDate = new Timestamp(System.currentTimeMillis());

File: src/main/java/com/example/TicketDAO.java [hunk 5]
```diff
@@ -1,54 +1,59 @@
...
-    public void saveTicket(Ticket ticket) throws SQLException {
-        try (Connection conn = DatabaseConnection.getConnection();
-             PreparedStatement stmt = conn.prepareStatement(SAVE_TICKET_SQL)) {
-
-            stmt.setInt(1, ticket.getUserId());
-            stmt.setString(2, ticket.getTicketType().name());  // Make sure to match enum name
-
-            stmt.executeUpdate();
+    public void saveTicket(Ticket ticket) {
+        //EntityManager em = emf.createEntityManager();
```
Comment (Alexmansar) [inline src/main/java/com/example/TicketDAO.java]: remove please

File: src/main/java/com/example/TicketDAO.java [hunk 6]
```diff
@@ -1,54 +1,59 @@
...
+        try(EntityManager em = emf.createEntityManager()) {
+            em.getTransaction().begin();
+            em.persist(ticket);
+            em.getTransaction().commit();
+            System.out.println("Saved new ticket for user ID: " + ticket.getUserId() + " with type: " + ticket.getTicketType());
+        } catch (Exception e) {
+            {
+                System.err.println("Error saving ticket: " + e.getMessage());
+                e.printStackTrace();
+            }
```
Comment (Alexmansar) [inline src/main/java/com/example/TicketDAO.java]: Why do we need this block?
I think **{** (21 line) and **}** (24 line) are redundant.

File: src/main/java/com/example/UserBase.java [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+@Table(name = "app_user")
+public abstract class UserBase implements User {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private int id;
+
+    @Column(name = "name", nullable = false)
+    private String name;
+
+    // Constructors, Getters, Setters
```
Comment (Alexmansar) [inline src/main/java/com/example/UserBase.java]: not needed

**Review by Alexmansar — CHANGES_REQUESTED** [review]
@gujaKhela Hi)) chech some comments please

**Review by Alexmansar — APPROVED** [review]
@gujaKhela  Could be merged))

**Pull request merged**